pkg/infra/provider.go

```diff
@@ -109,36 +111,43 @@ func (p *AWSProvider) Clean(ctx context.Context, cluster *infrav1.Cluster) error
 	return nil
 }
 
-func (p *AWSProvider) IsInitialized(ctx context.Context) bool {
+func (p *AWSProvider) IsInitialized(ctx context.Context) error {
 	capiCluster := &capiv1.Cluster{}
 	if err := p.Kube.Get(ctx, types.NamespacedName{Namespace: p.scope.Namespace, Name: p.scope.Name}, capiCluster); err != nil {
-		return false
+		return fmt.Errorf("failed to get Cluster: %v", err)
 	}
 
-	kcp := &controlplanev1.KubeadmControlPlane{}
-	if err := p.Kube.Get(ctx, types.NamespacedName{Namespace: capiCluster.Namespace, Name: capiCluster.Spec.ControlPlaneRef.Name}, kcp); err != nil {
-		return false
+	if conditions.IsFalse(capiCluster, capiv1.ReadyCondition) {
+		// merge all false conditions
+		errs := []error{}
+		for _, condition := range capiCluster.Status.Conditions {
+			if condition.Status == corev1.ConditionTrue || condition.Type == capiv1.ReadyCondition {
+				continue
+			}
+
+			errs = append(errs, errors.New(condition.Message))
```

Comment:
This is not right, there is case when the provider is working, and in that case I think you return nil error L125.

Then you will mark cluster as `InfrastructureProviderProvisionedCondition`